identity_credential/src/credential/proof.rs

```diff
@@ -0,0 +1,96 @@
+// Copyright 2020-2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use identity_core::common::Object;
+use serde::Deserialize;
+use serde::Serialize;
+
+/// Represents a cryptographic proof that can be used to validate verifiable credentials and
+/// presentations.
+///
+/// This representation does not inherently implement any standard; instead, it
+/// can be utilized to implement standards or user-defined proofs. The presence of the
+/// `type` field is necessary to accommodate different types of cryptographic proofs.
+///
+/// Note that this proof is not related to JWT and can be used in combination or as an alternative
+/// to it.
+#[derive(Clone, Debug, PartialEq, Eq, Deserialize, Serialize)]
+pub struct Proof {
+  /// Type of proof.
+  #[serde(rename = "type")]
+  pub _type: String,
```

Comment:
```suggestion
  pub type_: String,
```

Don't use `_type`, that is an implicit "ignore this variable" instruction. We don't want the compiler to not inform us if that field is unused. Use `type_` instead.